### ISSUE TITLE
Weight engine should return a UserWarning if any weight variables are delimited sets

### ISSUE THREAD

**Issue by jamesrkg** [Quantipy/quantipy#177]
If the user accidentally defines a weight variable used in a weight scheme as a delimited set, when they run the engine they will get a ``ZeroDivisionError: float division by zero`` error.

Instead, we should introduce a verification step that provides a ``UserWarning`` when a ``scheme`` is added to an ``engine`` and the scheme uses weight target variables that are not either ``int`` or ``single``. A non-fatal ``UserWarning`` is the better option here because later we will add native support for using ``delimited set``-typed columns for weight targets.

The  error actually occurs here:

```python
    def rakeonvar(self, target):
        #start=1 : This assumes that survey data ALWAYS starts with 1 (eg, sex: male=1, female=2, undisclosed=3)
        #index is the enumerator (1, 2, 3) and weight is the weight value [50.0, 50.0, 0.0]
        for index, weight in enumerate(self.targets[target], start=1):
            df = self.dataframe[self.dataframe[target] == index]  # df is a subset of the dataframe for the current target
            index_array = (self.dataframe[target] == index)
            data = df[self.weight_column_name] * (weight / sum(df[self.weight_column_name]))
            self.dataframe.loc[index_array, self.weight_column_name] = data
```

At the line ``data = df[self.weight_column_name] * (weight / sum(df[self.weight_column_name]))``. A ``try/except`` block should be used here to catch a ``ZeroDivisionError`` error and prompt the user to check if they are using an incorrectly typed column.

**Comment by alextanski** [comment]
Is this also possible?

Before the algoirthm starts, the _check_targets() method checks the data type of the passed variable(s):

ValueError: Scheme "simple", group "_default_name_": Variable q2 has unsuitable type for Weighting.
Target variables must be of type integer / single categorical.

**Comment by jamesrkg** [comment]
Resolved by #212.
